### PR TITLE
gnomeExtensions.x11-gestures:  can't find Touchegg

### LINKED ISSUE #137621: GNOME Extension X11 Gestures hardcodes Touchegg path
### Describe the bug
`gnomeExtensions.x11-gestures` (from [nixpkgs](https://search.nixos.org/packages?channel=21.05&show=gnomeExtensions.x11-gestures&from=0&size=50&sort=relevance&type=packages&query=x11+gestures)) is a GNOME extension which searches for the package `touchegg` in `/usr`

Because it can't detect the installed package it sends a notification that the `touchegg` package must be installed, and subsequently does not inhibit the default gestures provided by the package. Also, there is no built-in way to tell it to use another path.

### Steps To Reproduce
1. Install the `gnomeExtensions.x11-gestures` and `touchegg` (Please see additional context) packages
2. Enable the X11 Gestures extension

### Expected behavior
The extension must recognize the actual path for `touchegg` in `/nix/store` and work properly

### Additional context
Since the `touchegg` package in the repos is extremely outdated (v1.1.1), I am using an [overlay](https://github.com/maydayv7/nixos-config/blob/master/overlays/touchegg.nix) to install the latest version v2.0.11 (see [#116393](https://github.com/NixOS/nixpkgs/pull/116393) for updating the package in the repos).

Also, [#137131](https://github.com/NixOS/nixpkgs/pull/137131) may be relevant

### Maintainers
@piegamesde Extensions maintainer
@JoseExposito Creator of the extension and `touchegg`

### Metadata

```console
 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.62, NixOS, 21.05.3021.8b0b81dab17 (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.15`
 - channels(v7): `"home-manager-21.05"`
 - channels(root): `"nixos-21.05.3021.8b0b81dab17"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

### PR DESCRIPTION
**Pull request by AdsonCicilioti**
###### Motivation for this change
Fix #137621 

gnomeExtensions.x11-gestures: can't find Touchegg, which is a dependency on it.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/desktops/gnome/extensions/extensionOverrides.nix [hunk 1]
```diff
@@ -47,4 +48,12 @@ super: lib.trivial.pipe super [
 
     meta.maintainers = with lib.maintainers; [ rhoriguchi ];
   }))
+
+  (patchExtension "x11gestures@joseexposito.github.io" (old: {
+    # Extension can't find Touchegg
+    # https://github.com/NixOS/nixpkgs/issues/137621
+    postPatch = ''
+      substituteInPlace "src/touchegg/ToucheggConfig.js" --replace "GLib.build_filenamev([GLib.DIR_SEPARATOR_S, 'usr', 'share', 'touchegg', 'touchegg.conf'])" "'${touchegg}/share/touchegg/touchegg.conf'"
```
Comment (thiagokokada) [inline pkgs/desktops/gnome/extensions/extensionOverrides.nix:56]:
```suggestion
      substituteInPlace "src/touchegg/ToucheggConfig.js" \
        --replace "GLib.build_filenamev([GLib.DIR_SEPARATOR_S, 'usr', 'share', 'touchegg', 'touchegg.conf'])" "'${touchegg}/share/touchegg/touchegg.conf'"
```

**Review by thiagokokada — COMMENTED** [review]
Almost good, just a cosmetic change because the line is too big.

**Comment by thiagokokada** [comment]
Please squash your commits.

**Comment by thiagokokada** [comment]
Result of `nixpkgs-review pr 148457` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>gnomeExtensions.x11-gestures</li>
  </ul>
</details>

**Review by thiagokokada — APPROVED** [review]

**Comment by thiagokokada** [comment]
CC @maydayv7.

**Comment by maydayv7** [comment]
Works fine for me!

**Review by JoseExposito — COMMENTED** [review]
Hi! Touchégg and this extension developer here.

I arrived a bit late to review this. While this solution works fine, the steps to find the system configuration file are:

> If $XDG_CONFIG_DIRS is set, check if the config is present in one of those directories. Otherwise, fallback to /etc/xdg, as in the spec:
> https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html
> Finally, fallback to `/usr/share/touchegg/touchegg.cong` for backwards compatibility.

https://github.com/JoseExposito/gnome-shell-extension-x11gestures/blob/master/src/touchegg/ToucheggConfig.js#L165

So, there is no need to patch the code, adding the right path to `$XDG_CONFIG_DIRS` should be enough.

**Comment by thiagokokada** [comment]
> So, there is no need to patch the code, adding the right path to `$XDG_CONFIG_DIRS` should be enough.

Sadly I don't think we can set `$XDG_CONFIG_DIRS` in a way that this would work for everyone. We can expose `$XDG_CONFIG_DIRS` on NixOS itself using a module or something, however it wouldn't work for users of Nix package manager. An alternative is to wrap the binary, however in this case it is JavaScript file and not an executable, so it wouldn't also work.

I don't see another way to do this properly in nixpkgs unless we patch the binary.

**Pull request merged**